src/main/java/tc/oc/pgm/map/NameCacheUtil.java

```diff
@@ -0,0 +1,122 @@
+package tc.oc.pgm.map;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.stream.JsonWriter;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.UUID;
+import org.bukkit.Bukkit;
+import tc.oc.pgm.api.PGM;
+
+public class NameCacheUtil {
+
+  private static final Gson GSON = new Gson();
+
+  private static List<UUID> unresolved = new ArrayList<UUID>();
```

Comment:
```suggestion
  private static Set<UUID> unresolved = new HashSet<UUID>();
```
Make this a set, otherwise there's no purpose to it since several maps with same author would be added multiple times here.